### PR TITLE
Unset and disable embedded browser option if MacOS Big Sur

### PR DESCRIPTION
**Pull request by helin24**
Screen Shot 2020-11-19 at 10 59 34 AM

Temporarily fixes https://github.com/flutter/flutter-intellij/issues/5057

### CODE REVIEW

File: src/io/flutter/settings/FlutterSettings.java [hunk 1]
```diff
@@ -269,11 +270,16 @@ public boolean isEnableHotUiInCodeEditor() {
   }
 
   public boolean isEnableEmbeddedBrowsers() {
-    return getPropertiesComponent().getBoolean(enableEmbeddedBrowsersKey, true);
+    // TODO(helinx): Change default value once JxBrowser is compatible with MacOS 11.
+    return getPropertiesComponent().getBoolean(enableEmbeddedBrowsersKey, !SystemInfo.isMacOSBigSur);
```
Comment (devoncarew) [inline src/io/flutter/settings/FlutterSettings.java:274]: I think the 2nd param to this method is the default value if none is set. If they've previously set a value here, they'll get that value, which I assume we won't want for big sur.

I'd suggest just changing the start of this method to something like `if (SystemInfo.isMacOSBigSur) return false;`

File: src/io/flutter/settings/FlutterSettings.java [hunk 2]
```diff
@@ -269,11 +270,16 @@ public boolean isEnableHotUiInCodeEditor() {
...
   }
 
   public void setEnableEmbeddedBrowsers(boolean value) {
-    getPropertiesComponent().setValue(enableEmbeddedBrowsersKey, value, true);
+    // TODO(helinx): Remove this and change default value below once JxBrowser is compatible with MacOS 11.
+    if (SystemInfo.isMacOSBigSur) {
```
Comment (devoncarew) [inline src/io/flutter/settings/FlutterSettings.java:279]: I don't think we'll need to change the behavior of the setEnableEmbeddedBrowsers() w/ the change to isEnableEmbeddedBrowsers() I suggested above.

**Review by devoncarew — COMMENTED** [review]
Having it disabled on big sure makes sense.

I have some comments on the FlutterSettings.java changes; also, it would be nice (but not critical), to change the tooltip for the settings checkbox on big sur to explain when the checkbox was disabled.

File: src/io/flutter/sdk/FlutterSettingsConfigurable.java [hunk 3]
```diff
@@ -305,6 +306,9 @@ public void reset() {
 
     myEnableEmbeddedBrowsersCheckBox.setSelected(settings.isEnableEmbeddedBrowsers());
 
+    // TODO(helinx): Remove this once JxBrowser is compatible with MacOS 11.
+    myEnableEmbeddedBrowsersCheckBox.setEnabled(!SystemInfo.isMacOSBigSur);
```
Comment (devoncarew) [inline src/io/flutter/sdk/FlutterSettingsConfigurable.java:310]: Perhaps also (conditionally on if we're on big sur):

```
myEnableEmbeddedBrowsersCheckBox.setToolTipText("This option not available...");
```

**Review by devoncarew — APPROVED** [review]

File: src/io/flutter/settings/FlutterSettings.java (hunk 2 above)
Comment (helin24) [reply src/io/flutter/settings/FlutterSettings.java:274]: I thought the setting would show up as checked but disabled, but it's always empty if disabled.

File: src/io/flutter/sdk/FlutterSettingsConfigurable.java (hunk 3 above)
Comment (helin24) [reply src/io/flutter/sdk/FlutterSettingsConfigurable.java:310]: Oh I didn't realize you could set the tool tip here! Thanks

**Review by stevemessick — APPROVED** [review]

**Pull request merged**